docs/architecture/accounts.md

```diff
@@ -1,211 +1,123 @@
-Accounts are basic building blocks representing a user or an autonomous smart contract.
+# Accounts
 
-For smart contracts, the go-to solution is account-based state. Miden supports expressive smart contracts via a Turing-complete language and the use of accounts.
+> The primary entities of the Miden protocol
 
-In Miden, an account is an entity that holds assets and defines rules about how to transfer these assets.
+## What is an account?
 
-## Account design
+In Miden, an `Account` represents an entity capable of holding assets, storing data, and executing custom code. Each `Account` is a specialized smart contract providing a programmable interface for interacting with its state and managed assets.
 
-In Miden, every account is a smart contract. The diagram below illustrates the basic components of an account.
+### Account type
```

Comment:
I would move both "Account type" and "Account storage mode" sections after the "Account core components" section.

Reply:
Added it in a:

`## Additional information`

section